### PR TITLE
Adding ChaCha20/Poly1305 Support

### PR DESCRIPTION
**Pull request by jackctj117**
Added to build ffi and added to cipher.py. Not exactly sure what to add in cipher.py and same goes for the test.

### CODE REVIEW

**Comment by jackctj117** [comment]
Added encrypt and decrypt test, didn't include stream functionality.

File: scripts/build_ffi.py [hunk 1]
```diff
@@ -215,7 +215,7 @@ def make_flags(prefix, fips):
 
         flags.append("--disable-md5")
         flags.append("--disable-sha224")
-        flags.append("--disable-poly1305")
+        flags.append("--enable-poly1305")
```
Comment (dgarske) [inline scripts/build_ffi.py:218]: `--enable-chacha`?

**Review by dgarske — DISMISSED** [review]

File: scripts/build_ffi.py (hunk 1 above)
Comment (jackctj117) [reply scripts/build_ffi.py:218]: Just a few lines up it's enabled

**Review by danielinux — APPROVED** [review]
Tested OK

**Pull request merged**